gtsam/base/types.h

```diff
@@ -230,3 +230,30 @@ namespace std {
 #ifdef ERROR
 #undef ERROR
 #endif
+
+namespace gtsam {
+
+  /**
+   * A trait to mark classes that need special alignment.
+   * This is required to make boost::make_shared and etc respect alignment, which is essential for the Python
+   * wrappers to work properly.
+   */
+#if __cplusplus < 201703L
```

Comment:
Is c++17 not backwards compatible with c++11? If so, is this not superfluous? If not, we would need to add CI to ensure compatibility with c++17, which we are not doing at the moment, and so this is probably not needed, either.

Reply:
I removed the ifdef, since we are doing this in our own namespace there is no need to.